base_layer/core/src/proof_of_work/monero_rx/helpers.rs

```diff
@@ -174,20 +209,37 @@ pub fn create_ordered_transaction_hashes_from_block(block: &monero::Block) -> Ve
         .collect()
 }
 
-/// Appends merge mining hash to a Monero block
-pub fn append_merge_mining_tag<T: AsRef<[u8]>>(block: &mut monero::Block, hash: T) -> Result<(), MergeMineError> {
+/// Inserts merge mining hash into a Monero block
+pub fn insert_merge_mining_tag_into_block<T: AsRef<[u8]>>(
+    block: &mut monero::Block,
+    hash: T,
+) -> Result<(), MergeMineError> {
     if hash.as_ref().len() != monero::Hash::len_bytes() {
         return Err(MergeMineError::HashingError(format!(
             "Expected source to be {} bytes, but it was {} bytes",
             monero::Hash::len_bytes(),
             hash.as_ref().len()
         )));
     }
+    // When we insert the merge mining tag, we need to make sure that the extra field is valid.
+    let mut extra_field = parse_extra_field_truncate_on_error(&block.miner_tx.prefix.extra);
```

Comment:
I think it's better to error here. If there is an error, you will remove the remaining data from the extra field anyway, silently truncating the extra field

Reply:
Yes, that would not hurt